cmd/shimdiag/list.go

```diff
@@ -1,25 +1,69 @@
 package main
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"text/tabwriter"
 
 	"github.com/Microsoft/hcsshim/internal/appargs"
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
 	"github.com/urfave/cli"
 )
 
 var listCommand = cli.Command{
 	Name:      "list",
 	Usage:     "Lists running shims",
 	ArgsUsage: " ",
-	Before:    appargs.Validate(),
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "pids",
+			Usage: "Shows the process IDs of each shim",
+		},
+	},
+	Before: appargs.Validate(),
 	Action: func(ctx *cli.Context) error {
+		pids := ctx.Bool("pids")
 		shims, err := findShims("")
 		if err != nil {
 			return err
 		}
+
+		getPid := func(shimStr string) (int32, error) {
+			shim, err := getShim(shimStr)
+			if err != nil {
+				return 0, err
+			}
+			defer shim.Close()
+
+			svc := shimdiag.NewShimDiagClient(shim)
+			resp, err := svc.DiagPid(context.Background(), &shimdiag.PidRequest{})
+			if err != nil {
+				return 0, err
+			}
+			return resp.Pid, nil
+		}
```

Comment:
I feel like this should just go in a separate helper function

Reply:
Sure, I just like using these little anonymous functions :)

Reply:
I personally only really like to use anonymous functions if they're really small or if they need a lot of arguments from the calling function and are still small :p